model/cluster_utility.go

```diff
@@ -169,6 +172,76 @@ func (h *UtilityGroupVersions) AsMap() map[string]*HelmUtilityVersion {
 	}
 }
 
+// UnmarshalJSON is a custom JSON unmarshaler that can handle both the
+// old Version string type and the new type. It is entirely
+// self-contained, including types, so that it can be easily removed
+// when no more clusters exist with the old version format.
+// TODO DELETE THIS
+func (h *UtilityGroupVersions) UnmarshalJSON(bytes []byte) error {
+	type innerUtilityGroupVersions struct {
```

Comment:
I think we do not need this new `struct`. We can use `UtilityGroupVersions` here.

Reply:
We can use `h` directly because we'll end in a stack overflow, since we will end up calling this method each time we call `json.Unmarshal`.

Reply:
Right. My bad. Can we try this?

```go
type innerUtilityGroupVersions UtilityGroupVersions
var versions = &innerUtilityGroupVersions{}
```

And to assign this `versions` into `h`, we can use `*h = *(*UtilityGroupVersions)(versions)`